drivers/sensor/ags10/ags10.c

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) 2023 Balthazar Deliers
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT aosong_ags10
+
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/byteorder.h>
+
+#include "ags10.h"
+
+LOG_MODULE_REGISTER(AGS10, CONFIG_SENSOR_LOG_LEVEL);
+
+static int ags10_write(const struct device *dev, uint8_t cmd, uint8_t* data, uint8_t bytes)
+{
+	const struct ags10_config *conf = dev->config;
+	uint8_t send_buf[6] = {0};
+
+	send_buf[0] = cmd;
+	memcpy(&send_buf[1], data, bytes);

```

Comment:
This can potentially overflow `send_buf` because the length (`bytes`) is not checked.

Update - looks like this function is actually not used which is why the tests failed, so just remove it.

Reply:
I don't really understand why it says the function is not used: it's used in `ags10_init()` which is supposed to be called from macro `AGS10_INIT` automatically, right ?